### PR TITLE
Android: Set SYSTEM_UI_FLAG_HIDE_NAVIGATION when going to Fullscreen …

### PR DESCRIPTION
**Pull request by auzden**
…mode

Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [ ] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [x] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

Hides the navigation bar on API >= 14. On Android >= API 19 when immersive sticky mode is enabled the navigation bar can be swiped to be shown by the user.

https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_HIDE_NAVIGATION

This PR is related to the issue #

## Tasks

* [ ] Tested on Linux
* [ ] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [x] Tested on Android

## How to test this PR?

Describe how to best test these changes. Please provide a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve) if possible, you can use the follow template as a start:

On line 82 on main.cpp of the Android SFML example, change the window constructor to be:
```sf::RenderWindow window(screen, "", sf::Style::Fullscreen);```

### CODE REVIEW

**Comment by eXpl0it3r** [comment]
Shouldn't there be some code to remove the flag when switching back to windowed mode or is that automatically the case due to having to call `window.create()` to change the mode?

**Comment by eXpl0it3r** [comment]
So calling create() with window styling will make it appear again correctly, right?

File: src/SFML/Main/SFMLActivity.cpp [hunk 1]
```diff
@@ -183,13 +183,11 @@ void ANativeActivity_onCreate(ANativeActivity* activity, void* savedState, size_
     loadLibrary("sfml-window", lJNIEnv, ObjectActivityInfo);
     loadLibrary("sfml-graphics", lJNIEnv, ObjectActivityInfo);
     loadLibrary("sfml-audio", lJNIEnv, ObjectActivityInfo);
-    loadLibrary("sfml-network", lJNIEnv, ObjectActivityInfo);
```
Comment (eXpl0it3r) [inline src/SFML/Main/SFMLActivity.cpp]: Why was that removed?

File: examples/android/build.gradle [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 buildscript {
-    
+
     repositories {
         google()
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.1.0'
```
Comment (eXpl0it3r) [inline examples/android/build.gradle]: What's the difference between 3.0 and 3.1?

File: examples/android/app/build.gradle [hunk 3]
```diff
@@ -4,8 +4,8 @@ android {
     compileSdkVersion 27
     defaultConfig {
         applicationId "org.sfmldev.android"
-        minSdkVersion 14
-        targetSdkVersion 19
+        minSdkVersion 19
+        targetSdkVersion 26
```
Comment (eXpl0it3r) [inline examples/android/app/build.gradle]: What impact does it have updating these versions?

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/[Bb]uild/
```
Comment (eXpl0it3r) [inline .gitignore]: Don't think we need that.

File: src/SFML/Main/SFMLActivity.cpp (hunk 1 above)
Comment (auzden) [reply src/SFML/Main/SFMLActivity.cpp]: Bleh, my bad. Forgot I had an open PR.

File: examples/android/build.gradle (hunk 2 above)
Comment (MarioLiebisch) [reply examples/android/build.gradle]: It's basically the build system plugin that's downloaded and used. It's not necessarily mandatory to change, but might improve compatibility.

File: examples/android/app/build.gradle (hunk 3 above)
Comment (MarioLiebisch) [reply examples/android/app/build.gradle]: they essentially define the minimum Android version the app supports together with the API/header version it's working with. In this case it's targeting for stuff as defined in API 26, but versions down to 19 are supported with necessary adjustments/workarounds for those.

File: .gitignore (hunk 4 above)
Comment (MarioLiebisch) [reply .gitignore]: the folder name is defaulted by Gradle, so I'm undecided here.

File: src/SFML/Main/SFMLActivity.cpp (hunk 1 above)
Comment (MarioLiebisch) [reply src/SFML/Main/SFMLActivity.cpp]: Just create an extra branch for the next PR. :)

**Comment by eXpl0it3r** [comment]
Superseded by #1554
